src/web_or_worker_shell_read.js

```diff
@@ -20,7 +20,19 @@
       return new Uint8Array(/** @type{!ArrayBuffer} */(xhr.response));
     };
   }
-
+#if USE_FETCH
+  readAsync = (url, onload, onerror) => {
+    fetch(url)
+    .then(response => {
+      if(response.ok) {
+        return response.arrayBuffer();
```

Comment:
Why not call `onload` here?

Reply:
`response.arrayBuffer();` returns a promise that resolves to a buffer, not a literal buffer.